### ISSUE TITLE
1.6.4 Does Not start: Cannot get/create entry with parent path [cache] because of an incompatible intermediary value of type

### ISSUE THREAD

**Issue by tbenr** [librespot-org/librespot-java#1011]
**Describe the bug**
when running `java -jar librespot-player-1.6.4.jar` it doesn't start even with freshly generated config.

1.6.3 works instead (same config)

my java version:

openjdk 21.0.4 2024-07-16
OpenJDK Runtime Environment (build 21.0.4+7-Ubuntu-1ubuntu224.04)
OpenJDK 64-Bit Server VM (build 21.0.4+7-Ubuntu-1ubuntu224.04, mixed mode, sharing)

**Screenshots/Stracktraces/Logs**
```
Exception in thread "main" com.electronwill.nightconfig.core.IncompatibleIntermediaryLevelException: Cannot get/create entry with parent path [cache] because of an incompatible intermediary value of type: class com.electronwill.nightconfig.core.SimpleCommentedConfig
	at com.electronwill.nightconfig.core.concurrent.StampedConfig.getOrCreateConfig(StampedConfig.java:429)
	at com.electronwill.nightconfig.core.concurrent.StampedConfig.set(StampedConfig.java:576)
	at com.electronwill.nightconfig.core.utils.ConfigWrapper.set(ConfigWrapper.java:25)
2024-10-31 12:25:19,062 TRACE FileConfiguration:148 - Added new entry to configuration file: cache.dir
	at com.electronwill.nightconfig.core.Config.set(Config.java:26)
	at xyz.gianlu.librespot.player.FileConfiguration.checkMissingKeys(FileConfiguration.java:149)
	at xyz.gianlu.librespot.player.FileConfiguration.checkMissingKeys(FileConfiguration.java:144)
	at xyz.gianlu.librespot.player.FileConfiguration.updateConfigFile(FileConfiguration.java:196)
	at xyz.gianlu.librespot.player.FileConfiguration.<init>(FileConfiguration.java:95)
	at xyz.gianlu.librespot.player.Main.main(Main.java:36)
```

**Version/Commit**
1.6.4

**Comment by Fummowo** [comment]
Having the same issue 😓

**Comment by iVolt1** [comment]
Same issue here too.

**Comment by iVolt1** [comment]
I should add that 1.63 still works ok for me.

**Comment by chommik** [comment]
I'm having the same problem, so I did a quick bisect to get some clues. It turns out, that d68d59b6046f1df1e0a69347965d1e02137542d2 is the one to blame – librespot-player runs fine with this commit reverted.

**Comment by devgianlu** [comment]
> I'm having the same problem, so I did a quick bisect to get some clues. It turns out, that d68d59b6046f1df1e0a69347965d1e02137542d2 is the one to blame – librespot-player runs fine with this commit reverted.

Yes, that seems to be the problem. I cannot understand _why_ tho. I am a bit rusty with my own code after so many years. 

Since this is not actively developed anymore I will probably downgrade and pin the version of the lib.

**Comment by chommik** [comment]
> I cannot understand why tho.

I think it's Hyrum's Law ;)

Jokes aside, thanks for merging this and maintaining the project!

**Comment by devgianlu** [comment]
I have released [v1.6.5](https://github.com/librespot-org/librespot-java/releases/tag/v1.6.5)

**Comment by iVolt1** [comment]
v1.65 is working for me now.  Thanks for taking the time to fix this. Currently librespot-java is the only multiroom Spotify Connect with metadata option for Home Assistant.  Your continued support is greatly appreciated while new options on Home Assistant are in the works.
